### ISSUE TITLE
pandoc: Cannot decode byte '\xf6': Data.Text.Internal.Encoding.streamDecodeUtf8With: Invalid UTF-8 stream

### ISSUE THREAD

**Issue by lzlcfd** [jgm/pandoc#5600]
I want to convert the tex file into a docx document using the following command:

pandoc -f latex -t docx -o master.docx --bibliography ./proposal.bib master.tex

but pandoc gave the following error:

pandoc: Cannot decode byte '\xf6': Data.Text.Internal.Encoding.streamDecodeUtf8With: Invalid UTF-8 stream

I am assured that the tex file is saved as the utf-8 format, but still the problem occurred.

**Comment by wilx** [comment]
Can you reduce your file to a smallest possible test case that exhibits the issue?

**Comment by mb21** [comment]
> I am assured that the tex file is saved as the utf-8 format

Just in case, try: https://pandoc.org/MANUAL.html#character-encoding

**Comment by lzlcfd** [comment]
> > I am assured that the tex file is saved as the utf-8 format
> 
> Just in case, try: https://pandoc.org/MANUAL.html#character-encoding

Thank you for your reply, I tried the command line as you suggested, but still failed.

**Comment by agusmba** [comment]
> Can you reduce your file to a smallest possible test case that exhibits the issue?

**Comment by jgm** [comment]
You must be using an older version of pandoc, because recent versions will give the byte position of the decoding error; that may help in tracking this down.  Your input must not be valid UTF-8.

**Comment by jgm** [comment]
Check the bib file too!

**Comment by mb21** [comment]
Closing until we have a reproducible example...

**Comment by tranner** [comment]
I ran into similar problems and thought I would give some more details for others searching for the same issue.

A (quite) minimal example of the error:

``` latex
\usepackage{amsaddr}
```

with `amsaddr` downloaded from https://www.ctan.org/tex-archive/macros/latex/contrib/amsaddr.

```
 $ pandoc --version
pandoc 2.7.3
Compiled with pandoc-types 1.17.5.4, texmath 0.11.2.2, skylighting 0.8.1
Default user data directory: /localhome/scstr/.local/share/pandoc or /localhome/scstr/.pandoc
Copyright (C) 2006-2019 John MacFarlane
Web:  http://pandoc.org
This is free software; see the source for copying conditions.
There is no warranty, not even for merchantability or fitness
for a particular purpose.
```

```
$ echo "\usepackage{amsaddr}" | pandoc -t native
pandoc: Cannot decode byte '\xe9': Data.Text.Internal.Encoding.streamDecodeUtf8With: Invalid UTF-8 stream
```

Clearly the iconv fix doesn't make any difference here in the way suggest at https://pandoc.org/MANUAL.html#character-encoding but a possible fix is to look into amsaddr.sty and remove the bad characters from line 9
```diff
- %% Copyright (C) 2006 by J�r�me Lelong <jerome.lelong@gmail.com>
+ %% Copyright (C) 2006 by Lelong <jerome.lelong@gmail.com>
```
Then we have:
```
$ echo "\usepackage{amsaddr}" | pandoc -t native
[RawBlock (Format "tex") "\\usepackage{amsaddr}"]
```

The bad characters are in a comment so perhaps you would hope that this does not cause problems. Either way, this is a possible fix for users.

**Comment by defwxyz** [comment]
Is there a way to just ignore the invalid part of the text and keep parsing ?

How to reproduce : 

```
pandoc https://www.google.com
pandoc: Cannot decode byte '\xe9': Data.Text.Internal.Encoding.decodeUtf8: Invalid UTF-8 stream

```
I use latest version on debian linux buster
```
pandoc --version
pandoc 2.11.4
Compiled with pandoc-types 1.22, texmath 0.12.1, skylighting 0.10.2,
citeproc 0.3.0.5, ipynb 0.1.0.1
```

**Comment by jgm** [comment]
Well, I looked into this.  Despite using a meta tag that says it's UTF-8, the Google page is not UTF-8 encoded, nor is it served with a UTF-8 mime type (curl reports content type is `text/html; charset=ISO-8859-1`).
There is exactly one non-ASCII character, an `a0` between "Advertising" and "Program".  a0 is the unicode code point for nonbreaking space, but it is not how you encode a nonbreaking space into bytes with UTF-8.

Bottom line: this isn't a bug in pandoc. It's a bug in Google's home page.

**Comment by agusmba** [comment]
While I agree, Google's homepage does not follow [W3C recommendations regarding encoding](https://www.w3.org/International/questions/qa-html-encoding-declarations) (headers and meta should be coherent), it seems that HTTP Headers take precedence, and the headers state `ISO-8859-1`, so browsers treat Google's home page text as ISO... instead of UTF8.

Getting encoding right is always a bit messy (as you can see from Google itself)

**Comment by jgm** [comment]
Yes, I know this about browsers.  But pandoc only accepts UTF-8 input, as documented.  So because the page is not UTF-8, it doesn't work...as expected.  You can use `curl` and `iconv` in cases like this.

We could actually make this work more smoothly, I guess, by modifying `readURI` in Text.Pandoc.App.
Right now it's
``` hs
readURI :: FilePath -> PandocIO Text
readURI src = UTF8.toText . fst <$> openURL (T.pack src)
```
But since openURL returns a mime type, we could use that to try to determine how to decode the bytestring and use relevant functions in Data.Text.Encoding.  If you think this is a good idea, perhaps a new issue should be opened.

**Comment by jgm** [comment]
I made this change. So now we should be okay with at least latin1 and utf-8 encodings.
